torchtext/csrc/vectors.cpp

```diff
@@ -0,0 +1,68 @@
+#include <string>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct Vectors : torch::CustomClassHolder {
+private:
+  std::unordered_map<std::string, torch::Tensor> stovectors_;
+
+public:
+  // tokens_, vectors_, and unk_tensor_ holds the serialized params passed in
+  // during initialization. We need this because we need to be able to serialize
+  // the model so that we can save the scripted object. Pickle will get the
+  // serialized model from these members, thus they needs to be public.
+  std::vector<std::string> tokens_;
+  std::vector<torch::Tensor> vectors_;
+  torch::Tensor unk_tensor_;
+
+  explicit Vectors(std::vector<std::string> tokens,
+                   std::vector<torch::Tensor> vectors, torch::Tensor unk_tensor)
+      : tokens_(tokens), vectors_(vectors), unk_tensor_(unk_tensor) {
+    for (std::size_t i = 0; i < tokens.size(); i++) {
+      // guarding against size mismatch of vectors and tokens
+      if (i >= vectors.size()) {
```

Comment:
Why do this on every loop instead of looking at tokens.size() vs. vectors.size() at the beginning? Is this to detect duplicates? If so, why not at the end?

Reply:
Moved this check to the top and now throwing a `runtime_error`